Add RepeatCharactersRule.

amichair opened the pull request:
This rule is for determining if a password contains multiple sequences of repeating characters, which the RepeatCharacterRegexRule cannot match. For example, it might be legal for a password to contain a character that repeats twice (e.g. "express1"), but illegal to create a password with multiple doubling characters (e.g. "11223344").

Of course, for the specific case of sequence count = 1, this covers also what RepeatCharacterRegexRule checks, but more efficiently (without regexes).

dfish3r commented:
Just curious if this rule is mostly an academic exercise or will you be using this in a production capacity? Explaining complexity rules to users is often harder than implementing the rule. This one looks like it could frustrate some users.

Implementation looks good though.

amichair commented:
Rebased and refactored to use the new simpler RuleResult API and simplify the code a bit.

Yes, this is a real requirement to comply with a bank's password policy. I agree that it's confusing to the user, and moreover, like various other rules, is silly and does not increase security (and might actually hurt it), but whatcha gonna do :-)
